### PR TITLE
create library for evaluation

### PR DESCRIPTION
**Pull request by nthain**

### CODE REVIEW

File: src/eval_tool.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
+import cPickle
```
Comment (jetpack) [inline src/eval_tool.py]: Include the future copypasta?

```
from __future__ import absolute_import
from __future__ import division
from __future__ import print_function
```

File: src/eval_tool.py [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    def __init__(self, 
+                 model_version = None, 
+                 model_dir = '../models'):
+        self.model_dir = model_dir
+        if model_version:
+            self.load_model_and_tokenizer(model_version)
+            
+    def load_model_and_tokenizer(self, model_version):
+        self.model = load_model(os.path.join(self.model_dir, '%s_model.h5' % model_version))
+        self.tokenizer = cPickle.load(open(os.path.join(self.model_dir, '%s_tokenizer.pkl' % model_version), 'rb'))
```
Comment (jetpack) [inline src/eval_tool.py]: Nit: wrap this long line?

File: src/eval_tool.py [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+        
+    def prep_data(self, data_path, text_column, label_column = None, max_sequence_length = 1000):
+        """Loads the data from a csv.
+        
+        Args:
+            data_path: path to a csv file of the data to evaluate on.
+            text_column: column containing comment text in the csv.
+            label_column: column containing toxicity label in the csv. If set to None, it is assumed the data is unlabelled.
```
Comment (jetpack) [inline src/eval_tool.py]: Nit: wrap this long line?

File: src/eval_tool.py [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+from keras.preprocessing.text import Tokenizer
+from keras.preprocessing.sequence import pad_sequences
+from keras.utils import to_categorical
+from sklearn import metrics
+
+def compute_auc(y_true, y_pred):
+    fpr, tpr, thresholds = metrics.roc_curve(y_true, y_pred)
+    return metrics.auc(fpr, tpr)
+
+class EvalTool():
```
Comment (jetpack) [inline src/eval_tool.py]: What do you think about naming this "Model"?

File: src/eval_tool.py (hunk 3 above)
Comment (jetpack) [inline src/eval_tool.py]: Mention what this method returns?

File: src/eval_tool.py [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+        text_labels = None
+        if label_column:
+            text_labels = to_categorical(data[label_column])
+            
+        return text_data, text_labels
+            
+    def predict(self, data):
+        return self.model.predict(data)
+    
+    def score_auc(self, labels, preds):
```
Comment (jetpack) [inline src/eval_tool.py]: I don't think this needs to be a method - it doesn't use any object state.

Could we instead do the `[:,1]` earlier? For example, do it for `labels` in the `prep_data` method and for `preds` in the `predict` method?

File: src/eval_tool.py [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+    fpr, tpr, thresholds = metrics.roc_curve(y_true, y_pred)
+    return metrics.auc(fpr, tpr)
+
+class EvalTool():
+    def __init__(self, 
+                 model_version = None, 
+                 model_dir = '../models'):
+        self.model_dir = model_dir
+        if model_version:
+            self.load_model_and_tokenizer(model_version)
```
Comment (jetpack) [inline src/eval_tool.py]: Is there a usecase for this class if we don't pass model_version?

I think we could just make model_version required. We could also inline `load_model_and_tokenizer` into `__init__`. (This is more inline with making this a `Model` class.)

File: src/eval_tool.py (hunk 1 above)
Comment (nthain) [reply src/eval_tool.py]: Sure. Is it just to be safe?

File: src/eval_tool.py (hunk 4 above)
Comment (nthain) [reply src/eval_tool.py]: good idea. Done.

File: src/eval_tool.py (hunk 6 above)
Comment (nthain) [reply src/eval_tool.py]: I just wanted to provide options, but you're right. I've simplified.

File: src/eval_tool.py (hunk 2 above)
Comment (nthain) [reply src/eval_tool.py]: Done.

File: src/eval_tool.py (hunk 3 above)
Comment (nthain) [reply src/eval_tool.py]: Done.

Comment (nthain) [reply src/eval_tool.py]: Done.

File: src/eval_tool.py (hunk 5 above)
Comment (nthain) [reply src/eval_tool.py]: Ok. Though, this will cause problems if we unify training and evaluation. Done for now.

**Comment by nthain** [comment]
Responded to review in-line

File: src/eval_tool.py (hunk 1 above)
Comment (jetpack) [reply src/eval_tool.py]: Yeah, in particular the division thing. I often forgot to copy paste these imports, and things work. Later I expand the script/notebook, and happen to use division, expecting it work like the __future__, and get weird bugs.
